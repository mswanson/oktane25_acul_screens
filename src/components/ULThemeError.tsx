import * as React from "react";

import { Alert, AlertTitle } from "@/components/ui/alert";
import { cn } from "@/lib/utils";

export interface ULThemeErrorProps extends React.ComponentProps<"div"> {
  variant?: "destructive";
}

function ULThemeAlert({
  variant = "destructive",
  className,
  ...props
}: ULThemeErrorProps) {
  // Variant-specific theme overrides for Auth0 styling
  const variantThemeOverrides = {
    destructive: cn(
      "p-4",
      "theme-universal:bg-error", //background-color
      "theme-universal:text-(--ul-theme-color-primary-button-label)", //text-color
      "theme-universal:rounded-button", //border-radius
      "theme-universal:font-body", //font-weight
      "theme-universal:text-(length:--ul-theme-font-body-text-size)" //font-size
    ),
  };

  const themeClasses = cn(variant && variantThemeOverrides[variant]);

  return (
    <Alert
      variant={variant}
      className={cn(className, themeClasses)}
      {...props}
    />
  );
}

function ULThemeAlertTitle({
  className,
  ...props
}: React.ComponentProps<"div">) {
  // Theme overrides for error title styling
  const themeClasses = cn(
    "theme-universal:text-(--ul-theme-color-primary-button-label)", //text-color
    "theme-universal:font-body", //font-weight
    "theme-universal:text-(length:--ul-theme-font-body-text-size)" //font-size
  );

  return <AlertTitle className={cn(className, themeClasses)} {...props} />;
}

export { ULThemeAlert, ULThemeAlertTitle };
export default ULThemeAlert;
