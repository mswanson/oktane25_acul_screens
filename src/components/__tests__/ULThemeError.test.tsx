import { render, screen } from "@testing-library/react";

import { ULThemeAlert, ULThemeAlertTitle } from "../ULThemeError";

describe("ULThemeAlert", () => {
  // Snapshot test for component structure
  it("matches snapshot with ULThemeAlertTitle", () => {
    const { container } = render(
      <ULThemeAlert>
        <ULThemeAlertTitle>Error message</ULThemeAlertTitle>
      </ULThemeAlert>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  // Core functionality test
  it("renders as alert with proper accessibility", () => {
    render(
      <ULThemeAlert>
        <ULThemeAlertTitle>Test error</ULThemeAlertTitle>
      </ULThemeAlert>
    );

    const alert = screen.getByRole("alert");
    expect(alert).toBeInTheDocument();
    expect(screen.getByText("Test error")).toBeInTheDocument();
  });

  // Auth0 theme integration test
  it("applies Auth0 theme classes", () => {
    render(
      <ULThemeAlert data-testid="themed-alert">
        <ULThemeAlertTitle>Themed error</ULThemeAlertTitle>
      </ULThemeAlert>
    );

    const alert = screen.getByTestId("themed-alert");
    expect(alert.className).toContain("theme-universal:bg-error");
    expect(alert.className).toContain(
      "theme-universal:text-(--ul-theme-color-primary-button-label)"
    );
  });
});

describe("ULThemeAlert Integration", () => {
  // Real-world usage test matching IdentifierForm implementation
  it("handles multiple error scenarios like IdentifierForm", () => {
    const errors = [
      { message: "Enter a valid phone number" },
      { message: "Enter a valid phone number2" },
    ];

    render(
      <div className="space-y-3">
        {errors.map((error, index) => (
          <ULThemeAlert key={index}>
            <ULThemeAlertTitle>{error.message}</ULThemeAlertTitle>
          </ULThemeAlert>
        ))}
      </div>
    );

    expect(screen.getByText("Enter a valid phone number")).toBeInTheDocument();
    expect(screen.getByText("Enter a valid phone number2")).toBeInTheDocument();
  });
});
