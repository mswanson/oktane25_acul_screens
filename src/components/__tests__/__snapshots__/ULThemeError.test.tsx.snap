// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ULThemeAlert matches snapshot with ULThemeAlertTitle 1`] = `
<div
  class="shadow-input-hover bg-input relative grid w-full grid-cols-[0_1fr] items-start gap-y-0.5 overflow-clip rounded-3xl text-sm has-[>svg]:grid-cols-[calc(var(--spacing)*4)_1fr] has-[>svg]:gap-x-3 [&>svg]:size-4 [&>svg]:translate-y-0.5 text-destructive-foreground [&>svg]:text-destructive-foreground p-4 theme-universal:bg-error theme-universal:text-(--ul-theme-color-primary-button-label) theme-universal:rounded-button theme-universal:font-body theme-universal:text-(length:--ul-theme-font-body-text-size)"
  data-slot="alert"
  role="alert"
>
  <div
    class="col-start-2 line-clamp-1 flex h-auto min-h-4 leading-4.5 font-medium tracking-tight theme-universal:text-(--ul-theme-color-primary-button-label) theme-universal:font-body theme-universal:text-(length:--ul-theme-font-body-text-size)"
    data-slot="alert-title"
  >
    Error message
  </div>
</div>
`;
